lib/solr/query/request/facet.rb

```diff
@@ -58,7 +60,8 @@ def default_solr_h
               facet: subfacets.map(&:to_solr_h).reduce(&:merge),
               gap: gap,
               start: lower_bound,
-              end: upper_bound
+              end: upper_bound,
+              domain: facet_filter&.to_solr_h
```

Comment:
what's a domain?

Reply:
on solr docs they define the domain as the collection of documents returned by the query.
For some reason to apply a filter on the facets we have to use this syntax `{ domain: { filter: 'filter' } }`. 
https://lucene.apache.org/solr/guide/7_2/json-facet-api.html#FilteringFacets

I'll change this line to something more clear, like `domain: { filter: facet_filters } }`